netflix/src/components/main/sort/SortByOption.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from "react";
+import { filter } from "../../../redux/films/api";
+import { useSelector, useDispatch, batch } from "react-redux";
+import { setCurrentSort, setFilms } from "../../../redux/films/films";
+import { sortString } from "../../../helpers/sortHelperMethods";
+import { selectStatus } from "../../../redux/selectors/films";
+import { useSearchState } from "../../../hooks/useSearchState";
+
+const SortByOption = ({ option }: any) => {
+  const dispatch = useDispatch();
+  const searchState = useSelector(selectStatus);
+  const getQueryString = useSearchState();
+  const finalstring: string = sortString(option);
+  const [trigger] = filter(option);
+
+  const sortAll = async () => {
```

Comment:
Your code still can be optimized.

Imagine that in your `App.tsx` you have
```
const searchState = useSelector(selectStatus);
useEffect(() => {
    const queryString = getQueryString({
      currentGenre: searchState.currentGenre,
      currentSort: finalstring,
      currentSearch: searchState.currentSearch,
    });
    const payload = await trigger(queryString).unwrap();
    batch(() => {
      dispatch(setResults(payload.data.length));
      dispatch(setFilms(payload.data));
    });
}, [searchState]);
```

So in all places where you update `searchState` you simply need to update 1 _own_ state, e.g.
```
const sortAll = () => {
    const options = { finalstring, option };
    dispatch(setCurrentSort(options));
}
```